lib/Runtime/Base/ScriptContext.cpp

```diff
@@ -837,6 +837,18 @@ namespace Js
         return propertyRecord->GetPropertyId();
     }
 
+    void ScriptContext::GetOrAddPropertyRecord(Js::JavascriptString * propertyString, PropertyRecord const** propertyRecord)
+    {
+        if (VirtualTableInfo<Js::PropertyString>::HasVirtualTable(propertyString))
+        {
+            *propertyRecord = ((Js::PropertyString*)propertyString)->GetPropertyRecord();
```

Comment:
Are there any concerns about passing strings cross-contexts here? Should we require that the `propertyString` comes from this script context?

Reply:
good question and IMHO cross context here matters. I will add that.